### ISSUE TITLE
Defining a parameter as choice between a value and a range

### ISSUE THREAD

**Issue by Abrikosoff** [facebook/Ax#2495]
I have a use case where a particular variable should either be a fixed value, or be a value sampled from a fixed range; i.e., I would like to define a parameter of the form:
`{"name": "t_3", "type": "choice", "values": [0.0, RangeParameter(name='t_3', parameter_type=ParameterType.FLOAT, lower=8.0, upper=30.0)]},`

This gives me the error
`AssertionError: Values in `values` not of the same type and no `value_type` was explicitly specified; cannot infer value type for parameter t_3.`

Is there support in Ax for this use case, or is there an acceptable workaround?

Thanks in advance!

**Comment by mgarrard** [comment]
Hi @Abrikosoff, thanks for the question. We don't currently have support for this directly in Ax. Would be interested to learn more about your usecase to understand the context behind having the out of range default?

**Comment by Balandat** [comment]
Does "0" here represent that the parameter is inactive in some way? One could potentially express this as a hierarchical search space. 

A very crude workaround would be to simply use two parameters: A boolean `t_3_is_zero` and a range parameter `t_3` with the desired bounds, and then simply ignore the value of `t_3` in your evaluation if `t_3_is_zero` is `True`.

**Comment by Abrikosoff** [comment]
Hi @Balandat @mgarrard The concrete use case I have here is the modelling of an industrial process, where each input feature represents the execution (or not) of a particular step in this process. In the case of execution, a particular value needs to be attributed to that feature (as in, e.g., the setting or not of a external control parameter, and if yes, what is the continuous value of that parameter). Hence I think this does correspond to what @Balandat is suggesting, that the 0 setting is actually a state where that feature is not active. Would a hierarchical search space be helpful for this?

**Comment by Balandat** [comment]
Yes, a hierarchical search space is the right way to think about this. However, if you don't need to only return those parameter values that are "active" (and can do filtering on your end) then rather than defining a true hierarchical search space, the hack I suggested would essentially have the same effect (under the hood we currently flatten the hierarchical search spaces by default). So this may be the quicker way to get there.
